core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -387,6 +387,63 @@ public void testCompileNonRegisteredContext() throws IOException {
         }
     }
 
+    @Test
```

Comment:
Great test! Maybe one more that tests total compiles 2 or 3 or something - just more than one. So we can see how the number doesn't increment in weird ways.